src/java/org/apache/cassandra/index/sai/StorageAttachedIndexGroup.java

```diff
@@ -353,12 +353,12 @@ public boolean validateSSTableAttachedIndexes(Collection<SSTableReader> sstables
 
             if (indexDescriptor.isPerSSTableIndexBuildComplete())
             {
-                indexDescriptor.checksumPerSSTableComponents();
+                indexDescriptor.validatePerSSTableComponents(IndexValidation.CHECKSUM, true);
 
                 for (StorageAttachedIndex index : indexes)
                 {
                     if (indexDescriptor.isPerColumnIndexBuildComplete(index.identifier()))
-                        indexDescriptor.checksumPerIndexComponents(index.termType(), index.identifier());
+                        indexDescriptor.checksumPerIndexComponents(index.termType(), index.identifier(), validateChecksum);
                     else if (throwOnIncomplete)
                         throw new IllegalStateException(indexDescriptor.logMessage("Incomplete per-column index build"));
```

Comment:
nit: I know this falls into the category of non-essential clean up, but I'm realizing that the two exception messages created in this method, much like the ones in `IndexDescriptor` before you fixed them, are missing SSTable info. They could probably be fixed in much the same way.

Reply:
yeah I can fix this.